### PR TITLE
gsettings: add page

### PR DESCRIPTION
**Pull request by yump**
Examples for common uses of gsettings.  Uncommon uses, involving relocatable schemas and specifying their paths, are not included because I haven't run into them and do not understand them well enough.

- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

**Comment by CleanMachine1** [comment]
I can see quite a lot of basic errors which every first time contributor makes.

I'll try and work through it line by line. The more active you are, the quicker this will be since __you__ apply the edits and changes

File: pages/linux/gsettings.md
Comment (CleanMachine1) [inline pages/linux/gsettings.md:3]: What in the world is a schema validation, other than https://duckduckgo.com/?t=ffab&q=schema+validation 

Is that the average user will understand, because I don't think so.

File: pages/linux/gsettings.md
Comment (CleanMachine1) [inline pages/linux/gsettings.md:4]: This link is way to long I feel.

And from what I see, maybe we should use the manned.org link

File: pages/linux/gsettings.md
Comment (CleanMachine1) [inline pages/linux/gsettings.md:6]: By `dump` what do you mean, does it give complete raw data or is it more 'sincere' in how it gives the info

We normally use Display or Print

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md:3]: `gsettings set` will fail if you try to write a key that shouldn't exist, or supply a value that isn't in the range allowed by the schema. `dconf write` doesn't know about schemas and will let you write anything.

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md:4]: It is unfortunately very long, but the Redhat documentation does contain examples and links to learn more about the workings of dconf and gsettings.  Also, I feel that anyone who can `tldr gsettings` can just as well `man gsettings` on their own machine.  And may in fact have just *come* from the manpage.

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md]: I have made this change, using "display" where the output is more useful for human consumption, and "print" where the output is directly machine-readable and probably less likely to change and break one's scripts in the future.

Comment (yump) [reply pages/linux/gsettings.md]: It is indeed not entirely complete (doesn't contain the output of `range` or `describe`, for example), and not importable by anything AFAIK. But it is useful for piping to grep if you forget the exact name of a schema or key.

**Comment by yump** [comment]
Do you think it would be appropriate to add a verbatim example? like:

```
- Enable subpixel antialiased text rendering:

`gsettings set org.gnome.settings-daemon.plugins.xsettings antialiasing rgba`
```

or perhaps:

```
- Make caps lock work as an additional ctrl key

`gsettings set org.gnome.desktop.input-sources xkb-options "['ctrl:nocaps']"`
```

7 examples is already pushing it, and it comes close to running afoul of "write every example in a way that they cannot be unmindfully copy-pasted".

But the reason I looked for a tldr for gsettings in the first place was, "what is a VALUE supposed to look like anyway? what's a 'serialized GVariant'?"

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md]: I extended the `gsettings set` a little to mention this.

Comment (CleanMachine1) [reply pages/linux/gsettings.md]: I don't think its required or could just be shortened to `valid input` 

cc @SethFalco @navarroaxel

File: pages/linux/gsettings.md
Comment (CleanMachine1) [reply pages/linux/gsettings.md:4]: I'll look for an official site rather than a guide, if you can find one though since you know more about the topic, that would be better

File: pages/linux/gsettings.md
Comment (SethFalco) [reply pages/linux/gsettings.md]: I think most developers, sysadmins, and power users will probably understand or infer it.
It'd probably be fair to assume others might have to look it up.

I have no opinion on if it should be changed, though. Subjectively, I like it. ^-^'

Both are perfectly valid options, they're just different levels of explicitness:
`valid input` - Input is being validated _somehow_.
`schema validation` - Input is being validated against a schema. (A data model definition, like [JSON Schema](https://json-schema.org/) or [XSD](https://en.wikipedia.org/wiki/XML_Schema_(W3C)).)

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:3]: I'm not sure if we even need to be that specific. Reading that it's not obviously clear what it does (in my opinion). How about just:
> GSettings configuration tool

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:6]: Is there maybe an easier term or a way to describe `non-relocatable`?

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:10]: What does the `default` mean here? Is this behaviour default or does it use a default schema?

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:14]:
```suggestion
- Set the value of a key. Fails if the key doesn't exist or the value is out of range:
```

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:16]: Asking fellow maintainers: Should we use `example_key` here like we do everywhere or should we use the formatting style of a gsettings key (e.g. `always-show-log-out`)?

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:22]: Not sure about this comma here

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:26]:
```suggestion
- Display schema-allowed values for a key (helpful with enum keys):
```
Not sure this is needed. Saying that something's useful for one thing implies that it's not as useful for others.

File: pages/linux/gsettings.md
Comment (marchersimon) [reply pages/linux/gsettings.md:4]: It's unfortunate that it's so long, but it seems quite good and far more usable than the official GNOME documentation or the man page.

File: pages/linux/gsettings.md
Comment (marchersimon) [reply pages/linux/gsettings.md]: (https://github.com/tldr-pages/tldr/pull/6251#discussion_r673337616)

**Comment by marchersimon** [comment]
> Do you think it would be appropriate to add a verbatim example? like:
> 
> ```
> - Enable subpixel antialiased text rendering:
> 
> `gsettings set org.gnome.settings-daemon.plugins.xsettings antialiasing rgba`
> ```
> 
> or perhaps:
> 
> ```
> - Set keyboard to make caps lock work as ctrl, and right alt as compose:
> 
> `gsettings set org.gnome.desktop.input-sources xkb-options "['ctrl:nocaps', 'compose:ralt']"`
> ```
> 
> 7 examples is already pushing it, and it comes close to running afoul of "write every example in a way that they cannot be unmindfully copy-pasted".
> 
> But the reason I looked for a tldr for gsettings in the first place was, "what is a VALUE supposed to look like anyway? what's a 'serialized GVariant'?"

We **could** add one example, but would that bring any benefit? I think the examples we have now pretty much descibe how the command has to look like. If we add another discrete example it should be one you use all of the time.

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md:4]: One option might be the developer documentation for GSettings itself, which is part of GLib.  The url is not overlong: https://developer.gnome.org/gio/stable/settings.html.  It doesn't really talk about the command line tool `gsettings` at all, but AFAICT, `gsettings` is a fairly thin wrapper around a few of the functions there, and if you do a deep-dive on it you'll probably get a better understanding than the man page.

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md:10]: The default value for that key.

AFAICT, GSettings uses a dconf database backend that only stores changes from the default values specified in the schema definition.  Schema definitions are found in `/usr/share/glib-2.0/schemas`.

So:

```
$ gsettings reset org.gnome.settings-daemon.plugins.xsettings hinting
$ gsettings get org.gnome.settings-daemon.plugins.xsettings hinting
'slight'
$ dconf read /org/gnome/settings-daemon/plugins/xsettings/hinting
$ # (no output)
```

But if we explicitly set that key, even to one that matches the default...

```
$ gsettings set org.gnome.settings-daemon.plugins.xsettings hinting slight
$ gsettings get org.gnome.settings-daemon.plugins.xsettings hinting
'slight'
$ dconf read /org/gnome/settings-daemon/plugins/xsettings/hinting
'slight'
```

It is stored in the dconf database.

(And incidentally, we have triggered [a very old cairo bug](https://gitlab.freedesktop.org/cairo/cairo/-/issues/23).)

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md:6]: What I was trying to draw attention to is that `list-recursively` will not actually show *everything*.

What [the documentation](https://developer.gnome.org/gio/stable/GSettings.html) has to say about relocatable schemas is:

>A relocatable schema is one with no path attribute specified on its <schema> element.

[...]

>For example, a relocatable schema could be used to store geometry information for different windows in an application. If the schema ID was org.foo.MyApp.Window, it could be instantiated for paths /org/foo/MyApp/main/, /org/foo/MyApp/document-1/, /org/foo/MyApp/document-2/, etc.

[And elsewhere:](https://developer.gnome.org/gio/stable/gio-GSettingsSchema-GSettingsSchemaSource.html#g-settings-schema-get-path)

>Schemas may be single-instance or relocatable. Single-instance schemas correspond to exactly one set of keys in the backend database: those located at the path returned by this function.

>Relocatable schemas can be referenced by other schemas and can therefore describe multiple sets of keys at different locations. For relocatable schemas, this function will return `NULL`.

The main distinction seems to be that in order to operate on relocatable schemas, you need to know the GSettings schema id AND the dconf path.  `gsettings list-relocatable-schemas` can tell you the ids, but IDK how to get the paths where those schemas might be used.  [According to the discussion here](https://github.com/mate-desktop/mate-panel/issues/675), the `dconf-editor` GUI tool had a bear of a time trying to map the other way (from path to schema id).

It seems like most uses of the `gsettings` tool are, "paste this in your terminal to change such and such setting not exposed in the GUI," or new computer setup automation, which largely don't involve relocatable schemas.

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md:16]: For what it's worth, [the docs](https://developer.gnome.org/gio/stable/GSettings.html) do specifically say:

>Key names are restricted to lowercase characters, numbers and '-'.

**Comment by yump** [comment]
I looked it up and supposedly the Github UI lets you squash commits on merge if you want it to be a single commit, so I stopped squashing locally because it's error-prone and breaks the discussion history.

File: pages/linux/gsettings.md
Comment (yump) [reply pages/linux/gsettings.md]: In most recent version, I have re-ordered the examples to put the subcommands with scripting uses, `set`/`get`/`reset`, come first and the "exploratory" ones, including `list-recursively`, are farther down.  This makes the stuff about non-relocatable vs relocatable schemas less prominent, so hopefully users are less likely to get the impression they need to understand what that means to use the tool.

But I do think it should be there somewhere, as a warning that `list-recursively` does not display everything.

File: pages/linux/gsettings.md
Comment (marchersimon) [reply pages/linux/gsettings.md]: That's why I'm not sure. Independent of what the command uses, we generally use snake_case for arguments.

**Comment by marchersimon** [comment]
We squash PRs before merging by default

File: pages/linux/gsettings.md
Comment (marchersimon) [reply pages/linux/gsettings.md:4]: I don't think that page would be helpful for anyone wanting to use the command, but the `redhat` page still links to the official docs.

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:10]:
```suggestion
- Print the value of a key or the schema-provided default if the key has not been set in `dconf`:
```

File: pages/linux/gsettings.md
Comment (marchersimon) [inline pages/linux/gsettings.md:30]:
```suggestion
- Display the human-readable description of a key:
```

**Review by marchersimon — APPROVED** [review]
Looks good, thanks @yump

**Review by CleanMachine1 — APPROVED** [review]

**Review by navarroaxel — APPROVED** [review]

File: pages/linux/gsettings.md
Comment (SethFalco) [inline pages/linux/gsettings.md:14]: Sorry to be a pain!

```suggestion
- Unset a key, so that its schema default value will be used:
```

* https://www.collinsdictionary.com/dictionary/english/unset
* https://www.lexico.com/definition/unset
* https://www.merriam-webster.com/dictionary/unset

"Unset" is one word! Other than that I think this PR looks great!

**Review by SethFalco — APPROVED** [review]

**Pull request merged**
